### PR TITLE
removes margins from headings

### PR DESCRIPTION
**Pull request by matthewcarleton**
I've updated the headings to remove all margins. 

I've updated any instance of a heading to have the appropriate classes to get the desired styles throughout.

I've also moved the utilities from under components to be their own tab. This now includes a type section with Vertical Rhythm, Headings and Typography.

### CODE REVIEW

File: source/_patterns/utilities/typography/headings.hbs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<div class="container-fluid">
```
Comment (andresgalante) [inline source/_patterns/utilities/typography/headings.hbs]: Please remove everything from here, that's not the headings and also remove the `mb-pf-md` class. All the extra information should go just to the md file.

Leave just this:

```html
<h1 class="h1">Heading level 1</h1>
<h2 class="h2">Heading level 2</h2>
<h3 class="h3">Heading level 3</h3>
<h4 class="h4">Heading level 4</h4>
```

File: source/_patterns/utilities/typography/headings.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+title: Headings
+---
+## Overview
+For consistent typography throughout, we created heading helper classes.
+
+## Usage
+
+| Class | Usage |
+| -- | -- |
+| `.h1` **Applied to:** Any heading |  **Outcome:** Adds h1 sizing and weight **Required:** No |
```
Comment (andresgalante) [inline source/_patterns/utilities/typography/headings.md]: it can be applied to any element, even a p or a div

File: source/scss/_pf-type.scss [hunk 3]
```diff
@@ -51,22 +51,34 @@ h4 {
   @include pf-heading-styles($weight: $pf-font-weight-semi-bold);
 }
 
-.h1,
+.h1 {
```
Comment (andresgalante) [inline source/scss/_pf-type.scss]: Can we also define h5 and h6?

File: source/_patterns/utilities/typography/headings.hbs (hunk 1 above)
Comment (matthewcarleton) [inline source/_patterns/utilities/typography/headings.hbs]: Even the grid markup?

Comment (andresgalante) [inline source/_patterns/utilities/typography/headings.hbs]: everything, just leave this:
```html
<h1 class="h1">Heading level 1</h1>
<h2 class="h2">Heading level 2</h2>
<h3 class="h3">Heading level 3</h3>
<h4 class="h4">Heading level 4</h4>
```

**Pull request merged**